src/main/kotlin/com/vanniktech/maven/publish/MavenPublishPluginExtension.kt

```diff
@@ -7,31 +7,52 @@ import org.gradle.api.Project
  * @since 0.1.0
  */
 open class MavenPublishPluginExtension(project: Project) {
+
+  @JvmField
+  internal val defaultTarget = MavenPublishTarget(
+    project.findProperty("RELEASE_REPOSITORY_URL") as String? ?: System.getenv("RELEASE_REPOSITORY_URL") ?: "https://oss.sonatype.org/service/local/staging/deploy/maven2/",
+    project.findProperty("SNAPSHOT_REPOSITORY_URL") as String? ?: System.getenv("SNAPSHOT_REPOSITORY_URL") ?: "https://oss.sonatype.org/content/repositories/snapshots/",
+    project.findProperty("SONATYPE_NEXUS_USERNAME") as String? ?: System.getenv("SONATYPE_NEXUS_USERNAME"),
+    project.findProperty("SONATYPE_NEXUS_PASSWORD") as String? ?: System.getenv("SONATYPE_NEXUS_PASSWORD"))
+
+  @JvmField
+  internal val localTarget = MavenPublishTarget(
+    releaseRepositoryUrl = project.repositories.mavenLocal().url.toASCIIString(),
+    signing = false)
+
   /**
    * The release repository url this should be published to.
    * This defaults to either the RELEASE_REPOSITORY_URL Gradle property, the system environment variable or the sonatype url.
    * @since 0.1.0
    */
-  var releaseRepositoryUrl: String = project.findProperty("RELEASE_REPOSITORY_URL") as String? ?: System.getenv("RELEASE_REPOSITORY_URL") ?: "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+  var releaseRepositoryUrl: String
+    get() = defaultTarget.releaseRepositoryUrl
```

Comment:
Will we need these in the future?

Reply:
We can decide wether we want to keep the old properties for easy configuration of the default `uploadArchives` or want to deprecate them in favor of the new API in the next pr.

Reply:
Sounds good. Feel free to merge / move on.